kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/unique.kt

```diff
@@ -5,139 +5,251 @@ import io.kotest.matchers.MatcherResult
 
-fun FloatArray.shouldNotBeUnique(): FloatArray {
-   asList() shouldNot beUniqueByEquals("FloatArray")
-   return this
-}
+/**
+ * Asserts that the given [FloatArray] contains no duplicate elements using the equality
+ * method of the elements themselves.
+ *
+ * @return the input instance is returned for chaining
+ */
+fun FloatArray.shouldBeUnique(): FloatArray = apply { asList() should beUniqueByEquals("FloatArray") }
 
-fun DoubleArray.shouldBeUnique(): DoubleArray {
-   asList() should beUniqueByEquals("DoubleArray")
-   return this
-}
+/**
+ * Asserts that the given [FloatArray] contains duplicate elements using the equality
+ * method of the elements themselves.
+ *
+ * @return the input instance is returned for chaining
+ */
+fun FloatArray.shouldNotBeUnique(): FloatArray = apply { asList() shouldNot beUniqueByEquals("FloatArray") }
 
-fun DoubleArray.shouldNotBeUnique(): DoubleArray {
-   asList() shouldNot beUniqueByEquals("DoubleArray")
-   return this
-}
+/**
+ * Asserts that the given [DoubleArray] contains no duplicate elements using the equality
+ * method of the elements themselves.
+ *
+ * @return the input instance is returned for chaining
+ */
+fun DoubleArray.shouldBeUnique(): DoubleArray = apply { asList() should beUniqueByEquals("DoubleArray") }
 
 /**
- * Asserts that the given [Array] contains no duplicate elements using the equality
+ * Asserts that the given [DoubleArray] contains duplicate elements using the equality
  * method of the elements themselves.
  *
  * @return the input instance is returned for chaining
  */
-fun <T> Array<T>.shouldBeUnique(): Array<T> {
-   asList() should beUniqueByEquals("Array")
-   return this
-}
+fun DoubleArray.shouldNotBeUnique(): DoubleArray = apply { asList() shouldNot beUniqueByEquals("DoubleArray") }
+
+/**
+ * Asserts that the given [Array] contains no duplicate elements using the default equality
+ * method of the elements themselves.
+ *
+ * This assertion checks for uniqueness of elements based on their `equals` implementation.
+ *
+ * Example:
+ * ```
+ * val array = arrayOf("apple", "banana", "cherry")
+ * array.shouldBeUnique()   // Assertion passes
+ *
+ * val array = arrayOf("apple", "banana", "apple")
+ * array.shouldBeUnique()   // Assertion fails
+ * ```
+ *
+ * @return the input instance is returned for chaining.
+ */
+fun <T> Array<T>.shouldBeUnique(): Array<T> = apply { asList() should beUniqueByEquals("Array") }
 
 /**
- * Asserts that the given [Array] contains no duplicate elements using the given
+ * Asserts that the given [Array] contains no duplicate elements using the provided
  * [comparator] for equality.
  *
- * @return the input instance is returned for chaining
+ * This assertion uses the specified [comparator] to compare elements for uniqueness.
+ *
+ * Example:
+ * ```
+ * val array = arrayOf("apple", "banana", "APPLE")
+ * array.shouldBeUnique(String.CASE_INSENSITIVE_ORDER)   // Assertion passes
```

Comment:
should it be `CASE_SENSITIVE_ORDER`? because it's not unique when `"APPLE" == "apple"`